### ISSUE TITLE
Vectors:  Stroke/brush origin extra control points

### ISSUE THREAD

**Issue by RodneyBaker** [opentoonz/opentoonz#2861]
### Summary
All vector brushes appear to be especially vulnerable to distortion/end looping when Smoothing is set to greater than 33

![LeadingEdgeBrushError](https://user-images.githubusercontent.com/2576432/67628815-9e0d3200-f839-11e9-9a11-fffdb2f91428.png)
Note:  Strokes are being drawn from right to left.
The bump near the right is part of the vector brush.

### Steps to Reproduce
Create a vector level 
Select a vector brush from Style Editor (some will more easily display the leading edge of the stroke than others)
Draw vector lines
Note that the leading edge of the stroke will be distorted/looped or hooked over.
(Use the Iron Tool) to smooth it out)
Adjust Smoothness above and below the value of 33 and note the difference.

### Expected Results
Ends of vector line should run smoothing but directly into the control point.

### Actual Results
Leading edge of vector line hooks/loops over deforming stroke

--- 
Status
---
- [x] Known Issue
- [x] Documented Workaround(s)
- [ ] Documented Code Location(s)
- [ ] Priority Established
- [ ] Bounty Attached
- [ ] Developer Assigned/Attached

**Comment by zarco1794** [comment]
If a stroke is done quickly, smoothing has no effect. I don't know if this has to do with the issue raised here
![Animated GIF-source](https://user-images.githubusercontent.com/39812147/67642712-5a740000-f8ed-11e9-8445-3ca73ee4f0aa.gif)

**Comment by RodneyBaker** [comment]
@zarco1794  In your example it looks to me like you have Smoothing set to zero for both slow and fast drawing.  It is normal additional smoothing calculations are not triggered when Smoothness is set to zero.  Drawing a faster stroke does give us some generic smoothness (as seen in your example though so that is a good way to draw!)  Those experiencing brush lag are encouraged to turn Smoothness down to zero (if they can live with the results of course).  Shun Iwasawa described in another issue/topic more about the specifics of the calculations and their impact.

**Comment by RodneyBaker** [comment]
I was noticing this again today with regular vector brush while using touchpad on my laptop.
The result of the stroke being a concentrated and overlapped group of approx. 5 control points at the beginning of the stroke all almost hidden because they are overlaid over each other.

I note that running 'Replace Vectors with Simplified Vectors' resolves the problem (leaving only two control points in most cases).

While we wouldn't want intersecting lines to separate into segments as with normal use of 'Replace Vectors with Simplified Vectors' it seems to me running similar simplification code on the stroke as a last step (ignoring overlapping strokes if present) would not only resolve this issue but also many other issues where too many control points are created with vector drawing.

I'm posting this largely as a reminder for me to investigate the source code behind 'Replace Vectors with Simplified Vectors' to see if something there could actually be used to optimize drawing of vector strokes.  It would be important that the processing not slow down or cause unnecessary lag while drawing strokes.  And of course... hopefully there is an obvious reason why vector strokes are initiated with that odd hook at the origin and that itself can be addressed.

What today's discovery suggests to me is that it isn't just the custom vector brushes that has the unnecessary hook at the beginning of the stroke.

**Comment by RodneyBaker** [comment]
The primary workaround for this issue would appear to be to use the Iron Tool (or other tool such as Replace Vectors with Simplified Vectors) in order to resolve the extra/excess control points.

**Comment by RodneyBaker** [comment]
The odd hook at the origin of brush stroke also observed in #2526

**Comment by pojienie** [comment]
@RodneyBaker I want to share my discovery about "the hook" problem here:

First of all, like you, I can only reproduce this problem when I set smooth extra high (like 50) and draw really fast(slow drawing doesn't reproduce this problem). And the "hook" will only appear at the start of the stroke.

 The problematic code seems to be in here

https://github.com/opentoonz/opentoonz/blob/07d43670f9f1d4a42084e8bc0695be397e6c3f9d/toonz/sources/tnztools/toonzrasterbrushtool.cpp#L703-L705

I discovered two "solutions", by applying either "solution", the "hook" problem no longer appears for me (quotation because I am not sure if there is any side effect):

1. commenting out this part (remember to add variable declaration back as the code after this also uses those variables):

https://github.com/opentoonz/opentoonz/blob/07d43670f9f1d4a42084e8bc0695be397e6c3f9d/toonz/sources/tnztools/toonzrasterbrushtool.cpp#L703-L705

2. change

https://github.com/opentoonz/opentoonz/blob/07d43670f9f1d4a42084e8bc0695be397e6c3f9d/toonz/sources/tnztools/toonzrasterbrushtool.cpp#L704

to 

```
std::vector<TThickPoint>::iterator it = points.begin() + 1;
```
